### ISSUE TITLE
how to train the entity and relation models together?

### ISSUE THREAD

**Issue by lingzhiwisemed** [princeton-nlp/PURE#43]
Hi! How to train the entity and relation models together? Is there any relevant implementation code, thank you!

**Comment by a3616001** [comment]
Hi! I'm not pretty sure if I understand your question correctly, but our approach doesn't train the entity and relation models together. We need to train the entity model first to provide entity information for the relation model.

If you actually want a code to train two models sequentially, it should be straightforward to include training commands for both models in a bash script.

Let me know if you have further questions!

**Comment by lingzhiwisemed** [comment]
> Hi! I'm not pretty sure if I understand your question correctly, but our approach doesn't train the entity and relation models together. We need to train the entity model first to provide entity information for the relation model.
> 
> If you actually want a code to train two models sequentially, it should be straightforward to include training commands for both models in a bash script.
> 
> Let me know if you have further questions!

In Section 5.2, how to implement sharing of two representation encoders? thanks！

**Comment by a3616001** [comment]
Hi! To share the encoders during training, you can create training set combining entity training instances and relation training instances. At each iteration, random select a training batch from either entity instances or relation instances, compute the loss and update the shared model. 

I currently don't have a plan to release this code, but it should be straightforward to implement this based on the current code. Let me know if you need any further help!

**Comment by lingzhiwisemed** [comment]
Thank you!
